### PR TITLE
PSR-4 with virions support

### PR DESCRIPTION
**Pull request by Endermanbugzjfc**
# Makefile targets
- [X] `dev/plugin.yml`
- [X] `dev/resources`
- [X] `dev/src`: Convert PSR-4 source code structure to PSR-0.
# The poggit workflow
- [X] Dry-run php-cs-fixer.
- [X] PR merge converted source to poggit branch.
- [x] Change `psr-4` to `psr-0` in composer.json.
- [x] Uncomment `src-namespace-prefix` in plugin.yml.
- [X] Remind to update version in plugin.yml.
- [X] Trigger workflow by PR creation (to the poggit branch).

### CODE REVIEW

**Comment by Endermanbugzjfc** [comment]
又雙叒叕瘸了

**Comment by keopiwauyu** [comment]
bruh no bad words in commit i ned suqash merge now

File: .github/workflows/poggit.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        # encoding: 'utf8'            # Optional, defaults to 'utf8'
+        path: ./ # Optional, defaults to '.'
+    - uses: peter-evans/create-pull-request@v4
+      id: cpr
+      with:
+        base: poggit
+        token: ${{ secrets.GITHUB_TOKEN }}
+        commit-message: ${{ github.event.pull_request.body || 'PSR-4 to PSR-0 conversion workflow dispatch' }}
+        title: PSR-4 to PSR-0 conversion for poggit after merging ${{ github.head_ref || github.ref }}
+        branch: psr-420/${{ github.sha }}
```
Comment (keopiwauyu) [inline .github/workflows/poggit.yml:58]:
```suggestion
        branch: psr-420/${{ github.sha }}
        delete-branch: true
```

File: .github/workflows/poggit.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    - uses: thollander/actions-comment-pull-request@v1
+      if: contains(github.event_name, 'pull_request')
+      with:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        # ([] means conditional):
+        #
+        # - [~~]Merge #[?] into poggit.[~~]
+        # - Ensure the version number in plugin.yml / virion.yml is bumped.
+        # - Add repository to Poggit-CI.
+        # - Create the release.
```
Comment (keopiwauyu) [inline .github/workflows/poggit.yml:77]:
```suggestion
        # - Create the release.
        # - delete branck
```

File: .github/workflows/poggit.yml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+          - [ ] ${{ steps.param4comment.outputs.strikethrough || '' }}Merge #${{ steps.cpr.outputs.pull-request-number || '?' }} into poggit.${{ steps.param4comment.outputs.strikethrough || '' }}
+          - [ ] Ensure the version number in [plugin.yml](
+          ${{ github.server_url }}/${{ github.repository }}/blob/poggit/plugin.yml
+          ) / [virion.yml](
+          ${{ github.server_url }}/${{ github.repository }}/blob/poggit/virion.yml
+          ) is bumped.
+          - [ ] Add repository to [Poggit-CI](https://poggit.pmmp.io/ci).
+          - [ ] [Create](
+          ${{ github.server_url }}/${{ github.repository }}/releases/new
+          ) the release.
```
Comment (keopiwauyu) [inline .github/workflows/poggit.yml:90]:
```suggestion
          ) the release.
          - [ ] delete branck
```

File: Makefile [hunk 4]
```diff
@@ -39,9 +41,24 @@ phpstan-baseline.neon/regenerate: vendor
...
+	mkdir -p $(PARENT)
+	cp -r src $(PARENT)
+	mv $(PARENT)/src dev/src/$(SRC_NAMESPACE_PREFIX)
+
+dev/resources: resources
+	cp -rf resources dev/resources
+
+# dev/ExamplePlugin.phar: $(EXAMPLE_PLUGIN_SOURCE_FILES) dev/ConsoleScript.php $(EXAMPLE_PLUGIN_VIRIONS) dev/plugin.yml dev/src dev/resources
+dev/ExamplePlugin.phar: $(EXAMPLE_PLUGIN_SOURCE_FILES) dev/ConsoleScript.php $(EXAMPLE_PLUGIN_VIRIONS) dev/plugin.yml dev/src
 # 	$(PHP) dev/ConsoleScript.php --make plugin.yml,src,resources --out $@
```
Comment (keopiwauyu) [inline Makefile:59]:
```suggestion
```

**Review by keopiwauyu — CHANGES_REQUESTED** [review]
i fx suggestions

**Review by keopiwauyu — APPROVED** [review]

**Comment by keopiwauyu** [comment]
image
bruh first time contriutioner

**Pull request merged**